src/structure/bitindex.rs

```diff
@@ -17,6 +17,13 @@ use tokio::io::AsyncRead;
 /// The amount of 64-bit blocks that go into a superblock.
 const SBLOCK_SIZE: usize = 52;
 
+/// Calculate if it is a good idea to use a linear bitscan instead of the bitindex.
+/// We are assuming that this is the case if the start and end indexes are on the same cache line.
+#[inline(always)]
+fn use_linear_bitscan(start: usize, end: usize) -> bool {
```

Comment:
Why `usize` instead of `u64`? It seems like you have to add `as usize` to the args just to call `use_linear_bitscan`.